src/query.rs

```diff
@@ -191,4 +314,76 @@ mod tests {
             assert_eq!(props[props.len() - 2..], ["Version", "WindowsDirectory"])
         }
     }
+
+    #[test]
+    fn it_can_query_a_struct() {
+        let wmi_con = wmi_con();
+
+        #[derive(Deserialize, Debug)]
+        struct Win32_OperatingSystem {
+            Caption: String,
+        }
+
+        let results = wmi_con.query::<Win32_OperatingSystem>().unwrap();
+
+        for os in results {
+            assert_eq!(os.Caption, "Microsoft Windows 10 Pro");
+        }
+    }
+
+    #[test]
+    fn it_builds_correct_query_without_filters() {
+        #[derive(Deserialize, Debug)]
+        struct Win32_OperatingSystem {
+            Caption: String,
+        }
+
+        let query = build_query::<Win32_OperatingSystem>(None);
+        let select_part = r#"SELECT Caption FROM Win32_OperatingSystem "#.to_owned();
+
+        assert_eq!(query, select_part);
+    }
+
+    #[test]
+    fn it_builds_correct_query() {
+        #[derive(Deserialize, Debug)]
+        struct Win32_OperatingSystem {
+            Caption: String,
+        }
+
+        let mut filters = HashMap::new();
+
+        filters.insert("C1".to_string(), FilterValue::Str("a"));
```

Comment:
why can't "C1" accept 'static str?

Reply:
It can (in theory), but it's more code so I just defer it to when we would actually need it